src/redux/books/books.js

```diff
@@ -0,0 +1,87 @@
+const defaultState = [];
+
+// API data
+
+const APIURLPATH = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/j5lJCfJOKlFz5y9NieEJ/books';
+
+// Actions
+
+const FULLFILED = 'Bookshelf/books/FULLFILED';
+const ADD_BOOK = 'Bookshlef/books/ADD_BOOK';
+const REMOVE_BOOK = 'Bookshlef/books/REMOVE_BOOK';
+
+// Action creators
+
+export const fullfiled = (obj) => ({
+  type: FULLFILED,
+  payload: { obj },
+});
+
+export const addBook = (obj) => ({
+  type: ADD_BOOK,
+  id: obj.id,
+  title: obj.title,
+  author: obj.author,
+});
+
+export const removeBook = (id) => ({
+  type: REMOVE_BOOK,
+  id,
+});
+
+// Thunks
+
+export const fetchBooks = () => async (dispatch) => fetch(APIURLPATH)
+  .then((res) => res.json())
+  .then((data) => {
+    const dataArray = [];
+    Object.values(data).forEach((value) => {
+      dataArray.push(value[0]);
+    });
+    Object.keys(data).forEach((key, i) => {
+      dataArray[i].id = key;
+    });
+    dispatch(fullfiled(dataArray));
+  });
+
+export const postBookItem = (obj) => async (dispatch) => {
+  fetch(APIURLPATH, {
+    method: 'POST',
```

Comment:
- [ ] Good job so far. It is among the requirements to use `createAsyncThunk` here in redux file 😃 👍 
As a hint, your books reducer file should look like the following 👇
```js
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

Please implement it in your file too 😉